### PR TITLE
add exit handler to cleanup on failures

### PR DESCRIPTION
**Pull request by kahing**

### CODE REVIEW

File: test/small-integration-test.sh [hunk 1]
```diff
@@ -1,10 +1,16 @@
-#!/bin/bash -e
+#!/bin/bash -ex
```
Comment (andrewgaul) [inline test/small-integration-test.sh]: Intentional change?  If so can you use `set -o xtrace` instead (and `set -o errexit`).

File: test/small-integration-test.sh [hunk 2]
```diff
@@ -32,10 +38,10 @@ $S3FS $TEST_BUCKET_1 $TEST_BUCKET_MOUNT_POINT_1 \
     -o url=http://127.0.0.1:8080 \
     -o use_path_request_style
 
-./integration-test-main.sh $TEST_BUCKET_MOUNT_POINT_1
-
-fusermount -u $TEST_BUCKET_MOUNT_POINT_1
-
-kill $S3PROXY_PID
+if [ $? == 0 ]; then
```
Comment (andrewgaul) [inline test/small-integration-test.sh]: How does this ever false if `errexit` is set as above?

File: test/small-integration-test.sh [hunk 3]
```diff
@@ -1,10 +1,16 @@
-#!/bin/bash -e
+#!/bin/bash -ex
 
 # Require root
 REQUIRE_ROOT=require-root.sh
 #source $REQUIRE_ROOT
 source integration-test-common.sh
 
+function exit_handler {
+    kill $S3PROXY_PID
```
Comment (andrewgaul) [inline test/small-integration-test.sh]: Call `set +o errexit` here in case S3Proxy failed to start?

File: test/small-integration-test.sh (hunk 1 above)
Comment (kahing) [inline test/small-integration-test.sh]: done

File: test/small-integration-test.sh (hunk 2 above)
Comment (kahing) [inline test/small-integration-test.sh]: removed

File: test/small-integration-test.sh (hunk 3 above)
Comment (kahing) [inline test/small-integration-test.sh]: done

**Comment by kahing** [comment]
also amended commit to dump s3fs output to log file to ease debugging

File: test/small-integration-test.sh [hunk 4]
```diff
@@ -20,6 +48,8 @@ do
     sleep 1
 done
 
+S3PROXY_PID=$(netstat -lpnt | grep :8080 | awk '{ print $7 }' | sed -u 's|/java||')
```
Comment (andrewgaul) [inline test/small-integration-test.sh]: How about `lsof -i:8080 -t`?

**Comment by andrewgaul** [comment]
:+1:

**Pull request merged**